pandas/_libs/tslibs/period.pyx

```diff
@@ -1572,7 +1572,8 @@ cdef class _Period(object):
             code, stride = get_freq_code(freq)
             freq = get_freq_str(code, stride)
 
-        freq = to_offset(freq)
+        if not isinstance(freq, _BaseOffset):
```

Comment:
Could you do this check (using is_offset_object maybe) inside the cython to_offset function?

I’ll write about the larger cython issue when at a real keyboard.